app/views/products/_form.html.haml

```diff
@@ -1,28 +1,12 @@
-= form_with(model: @product, local: true) do |form|
-  %p
-    = form.label :title
-    %br/
-    = form.text_field :title
-  %p
-    = form.label :description
-    %br/
-    = form.text_area :description
-  %p
-    = form.label :price
-    %br/
-    = form.text_field :price
-    %p 
-    = form.file_field :photo
-    %br/
-  %p
-    = form.submit :save
-
-  - if @product.errors.any?
-    #error_explanation
-      %p
-        = pluralize(@product.errors.count, "error")
-        prohibited
-        this product from being saved:
-      %ul
-        - @product.errors.full_messages.each do |msg|
-          %li= msg
+%container
+  = render 'shared/error_messages_product'
+= form_with(model: @product, local: true) do |f|
+  = f.label :title
+  = f.text_field :title, class: 'form-control'
```

Comment:
leave new line between the label/text_fields groups, makes it easier to read :-)